File: frontend/src/api/orderApi.ts
import api from './axios';
export const orderApi = {
    create: (data: { shipping_address: string; notes?: string; items: { product_id: string; quantity: number }[] }) =>
        api.post('/orders', data),
    getMyOrders: (params?: Record<string, string>) => api.get('/orders', { params }),
    getAllOrders: (params?: Record<string, string>) => api.get('/admin/orders', { params }),
    getOrder: (id: string) => api.get(`/orders/${id}`),
    cancel: (id: string) => api.put(`/orders/${id}/cancel`),
    getSellerOrders: () => api.get('/sales'),
};

File: fluttermultiplatform/lib/api/order_api.dart
import 'api_client.dart';

class OrderApi {
  static Future create(Map<String, dynamic> data) => api.post('/orders', data: data);
  static Future getMyOrders({Map<String, dynamic>? params}) => api.get('/orders', queryParameters: params);
  static Future getAll({Map<String, dynamic>? params}) => api.get('/admin/orders', queryParameters: params);
  static Future getOrder(String id) => api.get('/orders/$id');
  static Future cancel(String id) => api.put('/orders/$id/cancel');
  static Future getMySales({Map<String, dynamic>? params}) => api.get('/sales', queryParameters: params);
}
